src/pip/_internal/operations/check.py

```diff
@@ -52,6 +56,10 @@ def check_package_set(package_set):
         missing_deps = set()  # type: Set[Missing]
         conflicting_deps = set()  # type: Set[Conflicting]
 
+        # Ignore dependency when it's not in the whitelist.
+        if whitelist is not None and package_name not in whitelist:
```

Comment:
For reasons of elegance, it would be nice if you didn't have to repeat the `None` check in each iteration of the loop. An alternative to passing a list or set would be to pass a `should_ignore()` function. The first couple lines of your function could then be:

```python
if should_ignore is None:
    should_ignore = lambda package_name: False
```

meaning your conditional would be:

```python
if should_ignore(package_name):
    continue
```